src/errors/asyncControllerHandler.ts

```diff
@@ -0,0 +1,10 @@
+import { NextFunction } from "express";
+
+// This method should be used to wrap any async controller methods to ensure error handling is applied
+export default async (next: NextFunction, method: () => void) => {
+    try {
+        await method();
+    } catch (error) {
+        next(error);
```

Comment:
What do we use next for in this app, if anything? As far as I can tell from reading about it, it's for falling back to other routes when the route has an error?

Reply:
It hands on to the next middleware which can handle the error. We need it for `handleError` to be triggered.